src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -295,6 +307,68 @@ public class HubEndPoint<THub, TClient> : EndPoint, IInvocationBinder where THub
             }
         }
 
+        private bool IsChannel(Type type, out Type payloadType)
+        {
+            var channelType = type.AllBaseTypes().FirstOrDefault(t => t.IsGenericType && t.GetGenericTypeDefinition() == typeof(ReadableChannel<>));
```

Comment:
It'll go away when we have `IAsyncEnumerable<T>`, because we can just cast and call `GetEnumerator`